pudb/debugger.py

```diff
@@ -1492,38 +1492,25 @@ def add_cmdline_content(s, attr):
                     coming_from="above")
 
         def cmdline_tab_complete(w, size, key):
-            from rlcompleter import Completer
+            try:
+                from jedi import Interpreter
+            except ImportError:
+                add_cmdline_content("Tab completion requires jedi to be installed.",
+                                    "command line output")
 
             text = self.cmdline_edit.edit_text
             pos = self.cmdline_edit.edit_pos
 
             chopped_text = text[:pos]
             suffix = text[pos:]
 
-            # stolen from readline in the Python interactive shell
-            delimiters = " \t\n`~!@#$%^&*()-=+[{]}\\|;:\'\",<>/?"
-
-            complete_start_index = max(
-                    chopped_text.rfind(delim_i)
-                    for delim_i in delimiters)
-
-            if complete_start_index == -1:
-                prefix = ""
-            else:
-                prefix = chopped_text[:complete_start_index+1]
-                chopped_text = chopped_text[complete_start_index+1:]
-
-            state = 0
-            chopped_completions = []
-            completer = Completer(cmdline_get_namespace())
-            while True:
-                completion = completer.complete(chopped_text, state)
-
-                if not isinstance(completion, str):
-                    break
-
-                chopped_completions.append(completion)
-                state += 1
+            try:
+                completions = Interpreter(chopped_text, [cmdline_get_namespace()]).completions()
+            except:
+                # Jedi sometimes produces errors. Ignore
```

Comment:
How frequent are those? Could we at least add an advisory error message to the console output if that happens? Because, you know, "errors should never pass silently"? :)

Reply:
I don't know what it's like these days. I can try printing them and we can see.

Reply:
I made it print the error. I don't know if jedi is actually better these days, so we'll see if it actually happens a lot or not. I also don't know to what degree it actually executes stuff like `dir()` on the objects in the namespace. The rlcompleter definitely does that, which can cause arbitrary code execution if the objects do certain things (like define `__dir__`).